public/request/achievement/update.php

```diff
@@ -18,6 +18,14 @@
 $field = requestInputPost('f', null, 'integer');
 $value = requestInputPost('v');
 
+$achievementTitle = requestInputPost('t');
+$desc = requestInputPost('d');
+$badgeName = requestInputPost('b');
+$achPoints = requestInputPost('p', null, 'integer');
+$achMem = requestInputPost('m');
+$gameID = requestInputPost('g', null, 'integer');
+$achFlags = requestInputPost('l');
```

Comment:
as these are all specific to "f=4", they should be in the if block for that (similar to 'g' for "f=1")

Reply:
I tried to split this out the best I could, doing so required splitting the jrDevAllowed check a bit more which I'm not fully happy with how it ended up as well as duplicate calls on gameID and achFlags in case 4 for Jrs but phpstan complains if I put those in if statements